graveler/pgrefs.go

```diff
@@ -169,6 +168,58 @@ func (m *PGRefManager) ListBranches(ctx context.Context, repositoryID Repository
 	return NewBranchIterator(ctx, m.db, repositoryID, IteratorPrefetchSize, string(from)), nil
 }
 
+func (m *PGRefManager) GetTag(ctx context.Context, repositoryID RepositoryID, tagID TagID) (*CommitID, error) {
+	commitID, err := m.db.Transact(func(tx db.Tx) (interface{}, error) {
+		var commitID CommitID
+		err := tx.Get(&commitID, `SELECT commit_id FROM graveler_tags WHERE repository_id = $1 AND id = $2`,
+			repositoryID, tagID)
+		if err != nil {
+			return nil, err
+		}
+		return &commitID, nil
+	}, db.ReadOnly(), db.WithContext(ctx))
+	if errors.Is(err, db.ErrNotFound) {
+		return nil, ErrNotFound
+	}
+	if err != nil {
+		return nil, err
+	}
+	return commitID.(*CommitID), nil
+}
+
+func (m *PGRefManager) SetTag(ctx context.Context, repositoryID RepositoryID, tagID TagID, commitID CommitID) error {
+	_, err := m.db.Transact(func(tx db.Tx) (interface{}, error) {
+		_, err := tx.Exec(`
+			INSERT INTO graveler_tags (repository_id, id, commit_id)
+			VALUES ($1, $2, $3)
+				ON CONFLICT (repository_id, id)
+				DO UPDATE SET commit_id = $3`,
```

Comment:
In git, tags are immutable. I think they can be force deleted,  but not updated

Reply:
Moving SetTag to be CreateTag - will require delete and create for force update.